socorro/tests/external/es/test_crashstorage.py

```diff
@@ -204,144 +213,91 @@ def test_index_crash_mapping_keys(self):
+        return build_instance(
+            class_path="socorro.external.es.crashstorage.ESCrashStorage",
+            kwargs=settings.CRASH_DESTINATIONS["es"]["options"],
         )
 
-        # Ensure that the indexing function is only called once.
-        assert es_storage._submit_crash_to_elasticsearch.call_count == 1
-
-    @mock.patch("socorro.external.es.connection_context.elasticsearch")
-    def test_success(self, espy_mock):
+    def test_indexing_success(self, es_helper):
         """Test a successful index of a crash report"""
+        crash_id = create_new_ooid()
         raw_crash = {
             "BuildID": "20200605000",
             "ProductName": "Firefox",
             "ReleaseChannel": "nightly",
         }
         processed_crash = {
-            "uuid": "936ce666-ff3b-4c7a-9674-367fe2120408",
-            "json_dump": {},
+            "uuid": crash_id,
             "date_processed": date_to_string(utc_now()),
             "dom_fission_enabled": "1",
+            # NOTE(willkg): json_dump is not a supersearch field, so it gets dropped
+            # when indexing
+            "json_dump": {},
         }
 
-        sub_mock = mock.MagicMock()
-        espy_mock.Elasticsearch.return_value = sub_mock
-
-        crash_id = processed_crash["uuid"]
-
         # Submit a crash like normal, except that the back-end ES object is
         # mocked (see the decorator above).
-        es_storage = ESCrashStorage(config=self.config)
-        es_storage.save_processed_crash(
+        crashstorage = self.build_crashstorage()
+        crashstorage.save_processed_crash(
             raw_crash=raw_crash,
             processed_crash=processed_crash,
         )
+        es_helper.refresh()
 
-        # Ensure that the ES objects were instantiated by ConnectionContext.
-        assert espy_mock.Elasticsearch.called
-
-        # Ensure that the IndicesClient was also instantiated (this happens in
-        # IndexCreator but is part of the crashstorage workflow).
-        assert espy_mock.client.IndicesClient.called
-
-        # The actual call to index the document (crash).
-        document = {
+        # Assert what got saved
+        doc = es_helper.get_crash_data(crash_id)
+        assert doc == {
             "crash_id": crash_id,
             "raw_crash": {},
             "processed_crash": {
-                "uuid": "936ce666-ff3b-4c7a-9674-367fe2120408",
-                "date_processed": string_to_datetime(processed_crash["date_processed"]),
+                "uuid": crash_id,
+                "date_processed": processed_crash["date_processed"],
                 "dom_fission_enabled": "1",
             },
         }
 
-        additional = {
-            "doc_type": "crash_reports",
-            "id": crash_id,
-            "index": self.es_context.get_index_for_date(utc_now()),
-        }
-
-        sub_mock.index.assert_called_with(body=document, **additional)
-
-    @mock.patch("socorro.external.es.connection_context.elasticsearch")
```

Comment:
This didn't test anything real because everything was mocked out so I removed it.